starter-code/routes/index.routes.js

```diff
@@ -1,7 +1,19 @@
-const express = require('express');
-const router = express.Router();
+require('dotenv').config()
+const express = require('express')
+const passport = require('passport')
+const router = express.Router()
+
+const User = require('../models/User.model.js')
+const Post = require('../models/Post.model.js')
+const uploadCloud = require('../configs/cloudinary.js')
+
+const { ensureLoggedIn, ensureLoggedOut } = require('connect-ensure-login')
 
 /* GET home page. */
-router.get('/', (req, res) => res.render('index', { title: 'Express - Generated with IronGenerator' }));
+router.get('/', async (req, res) => {
+  const allPosts = await Post.find({}).populate('creatorId', 'username')
```

Comment:
```suggestion
  const allPosts = await Post.find({}).populate('creatorId')
```
Populate recibe como parametro un string con el nombre de la propiedad que deseas popular del modelo en el cual estas buscando, para tu caso creatorId te funciona, esto hace que en vez de darte el id te devuelve un objeto con la informacion que esta en ese registro, si deseas el username lo que  necesitas es solo hacer algo como `post.creatorId.username` para cada post.